src/compas/datastructures/mesh/_mesh.py

```diff
@@ -644,17 +645,17 @@ def from_vertices_and_faces(cls, vertices, faces):
         """
         mesh = cls()
 
-        if type(vertices) == list:
+        if isinstance(vertices, collections.MutableSequence):
```

Comment:
can it not just be `collections.Sequence`?
doesn't have to be a list, can also be a tuple...

Reply:
I know this is merged already so I'm late to the party, but I think this change rules out iterators, which is a real pity. As suggested by @brgcode , I think best would be *duck typing* style, but if you want to keep the type check, please try with `collections.Iterable` and add a bunch of simple tests that make sure this works with the basic stuff: lists, tuples, iterators/generators, dicts, and ideally also a custom iterable class.

Reply:
i already changed this locally, will push it soon. i think the check for mapping is useful, but that should it. i already had problems with numpy arrays for example.